### DISCUSSION TITLE
Jest: warning shows about multiple jotai instances when used in tests with requireActual

### DISCUSSION THREAD

**Discussion by scamden** [pmndrs/jotai#2460] [Bug report] [answered]
[This warning](https://github.com/pmndrs/jotai/blob/419ab5cda3503e70463dc340076e21e438db89b6/src/vanilla/store.ts#L865) is firing when I used requireActual on a module that imports jotai. From debugging it seems straightforwardly that jest first imports the module which imports jotai in order to mock it. Then when the mock calls requireActual on that module it reexecutes the module and it's dependencies causing this warning to fire.

Not sure what the fix is, maybe we could NODE_ENV for 'test' as well?

**Comment by dai-shi** [comment]
Even if it's the test env, the warning is correct. There are two stores in the memory, and it may cause unexpected behaviors. If you are sure that you don't use old stores, you can ignore the warning with mocking console.log. Or, maybe you could mock `getDefaultStore`. Yeah, mocking `getDefaultStore` makes sense if it's possible.

**Reply by scamden** [reply]
Maybe this is nit picky but in this case I don't think 
> There are two stores in the memory

is true. 

The module initializes two times but `getDefaultStore` is only called one time. The warning still fires because the count is incremented on each module init rather than on each call to the function.

By mocking `getDefaultStore` did you mean mock it and cache my own default store and return that? Feels a little strange to do exactly what the function is already doing just to avoid a warning but I can try that if you think that's the right solution. I can also just ignore the warning but it's a bit messy.

**Reply by dai-shi** [reply]
> The module initializes two times but getDefaultStore is only called one time. 

Oh, that's nice catch!! Let's improve it so that it warns only if getDefaultStore returns a different instance. I will open a PR later.

**Reply by dai-shi** [reply, marked as answer]
#2462 is open.

https://ci.codesandbox.io/status/pmndrs/jotai/pr/2462
Can you try it please? Find "Local Install Instructions" ☝️

**Reply by scamden** [reply]
it works! no warning! thank you!

**Reply by dai-shi** [reply]
Nice. Thanks for the hint!
